lib/client-options.js

```diff
@@ -665,6 +666,18 @@ function setRustOptions(options) {
     rustOptions.applicationVersion = options.applicationVersion;
     rustOptions.keyspace = options.keyspace;
     rustOptions.cacheSize = options.maxPrepared;
+    // We can either:
+    // - not have policies,
+    // - have policies, but not have address resolution policy,
+    // - have a address resolution policy, that works as a policy, but is not inherited from the default policy (duck typing),
+    // - have a proper address resolution policy.
+    // In the first two cases, we treat that as lack of policy, and the third case as custom policy.
+    rustOptions.addressTranslationPolicy =
+        options.policies && options.policies.addressResolution
+            ? options.policies.addressResolution.isDefaultPolicy
+                ? options.policies.addressResolution.isDefaultPolicy()
+                : rust.AddressTranslatorPolicies.CustomPolicy
+            : null;
```

Comment:
:thinking: This looks like `isDefaultPolicy()` should rather be `makeDefaultPolicy()`, no?
`is*` suggests that the function should return boolean.

Reply:
Once again. This is definition of `isDefaultPolicy`. It does not return a boolean, so it's not a predicate. I don't understand why it should be called `is*` then.
```js
isDefaultPolicy() {
        if (
            Object.getPrototypeOf(this) === EC2MultiRegionTranslator.prototype
        ) {
            return rust.AddressTranslatorPolicies.EC2MultiRegion;
        }
        return rust.AddressTranslatorPolicies.CustomPolicy;
    }
```